Boons, Endorsements and Bugfixing

hex37 opened the pull request:
- Adds the Endorsements system; All vampire faction leaders now require a modest political base of other players in order to remain in power.
- Adds Boons; now you can track the favors other vampires owe you.
- Adds a new page to the character setup menu for tracking and deleting connections
- adds verbs for offering and requesting boons
- adds verbs for offering and requesting endorsements
- refactors the entire character connection system to be managed via subsystem
- Changes the start-of-game message to say if the Seneschal is or isn't present, instead of the Prince who is Always Watching, from the shadows.
- makes most non-tgui menus use HTML5
- Removed runtime regarding keybindings
- Added a vampire ankh backsplash to many non-tgui menus
- resized the join game window
- patches up some holes on the map
- Tightens up code for Obfuscate, adds more deactivation triggers
- Removes Generation code wholesale 
- refactors most NPC HP calculation to use the default human hp calculator
- Failing Diablerie no longer transfers the victim's mind to the diablerist, for consent reasons. Functionality may be readded later.
- Renames several disciplines to be better named
- Changes the typo in the entire naming structure of the resilience discipline
- Balances Celerity to only have jojo-punching at level 5
- reduces Dominate's sleep command to be shorter.
- increases protean claw damage
- max blood pool is no longer recalculated every freaking life tick (wtf)
- Alters the backend of the character connection Database to allow for larger strings on some tables.

hex37 committed:
boons and endorsements, sans round-start check

elgeonmb commented on code/modules/vtr13/connections/endorsements/endorsement_bishop.dm:
```diff
@@ -0,0 +1,5 @@
+	name = CONNECTION_ENDORSEMENT_BISHOP
+	required_faction = EXP_TYPE_LANCEA
+	desired_position = "Bishop"
+	minimum_endorser_rank = VAMP_RANK_NEONATE
```
all roles other than carthian rep should have minimum_candidate_rank at ancilliae

hex37 replied:
I will fix this

elgeonmb commented on code/modules/vtr13/connections/screen_alerts/alert_connection.dm:
```diff
@@ -0,0 +1,50 @@
+	if(!.)
+		return
+	if(!iscarbon(usr))
+		CRASH("User for [src] is of type \[[usr.type]\]. This should never happen.")
```
do gangrel lose their ability to interact with this system when they turn into dogs?

hex37 replied:
i hate transform code i hate transform code i hate -


Yes they should use their ability to interact with the system but I need to test it.

elgeonmb commented on code/modules/vtr13/connections/subsystems/SScharacter_connection_procs/db_operations/endorsements/check_can_endorse.dm:
```diff
@@ -0,0 +1,74 @@
+	
+	if(endorser_must_be_faction_head)
+		if(!SScharacter_connection.check_is_eligible_for_faction_head(endorser))
+			to_chat(endorser, span_notice("Your character cannot give this endorsement without being an eligible faction head for the Lancea et Sanctum, Ordo Dracul, Circle of the Crone, or the Carthian Movement."))
```
Invictus faction heads should probably also be allowed

hex37 replied:
I thought we agreed on it needing to be from other covenants? Otherwise it'll just be an internal invictus "hugbox" because they have enough head roles to promote internally.

elgeonmb commented on code/modules/vtr13/connections/subsystems/SScharacter_connection_procs/retire_connection.dm:
```diff
@@ -0,0 +1,15 @@
+	for(var/datum/character_connection/connection in target.mind.character_connections)
+		if(connection.group_id == deleting_group_id)
+			message_admins("[ADMIN_LOOKUPFLW(target)] has ended their [connection.group_type] connection of group_id [connection.group_id]. (They were a [connection.member_type])")
+			log_game("[key_name(target)] has ended their [connection.group_type] connection of group_id [connection.group_id]. (They were a [connection.member_type])")
```
Does this go to overwatch?

hex37 replied:
This does not

elgeonmb commented on code/modules/vtr13/mobs/human/handle_ghouling.dm:
```diff
@@ -0,0 +1,36 @@
+		return FALSE
+
+	var/datum/preferences/BLOODBONDED_prefs_g = src.client.prefs
+	if(BLOODBONDED_prefs_g.discipline_types.len == 3)
```
not actually sure if we want characters who are ghouled to get disciplines that night; it's not lore-compliant though it's probably more fun to let people play with their toys right away

hex37 replied:
Oh - yeah I can change that. This is leftover and transplanted code from the give vitae process.

hex37 committed:
I snuck in some discipline changes

it's horrible practice, I know

Also implements a backsplash on the preferences pages, and all common browse windows

Continues endorsement work

hex37 committed:
obfuscate disguises beast; darker preferences page

hex37 committed:
comment update

hex37 committed:
base endorsement code done

hex37 committed:
verb bugfixing

hex37 committed:
job fixing

hex37 committed:
more bugfixing

hex37 committed:
turn off query lockup

hex37 committed:
format fix

hex37 committed:
i fix code -smiles-

hex37 committed:
save faction correctly

hex37 committed:
fix minimum vamp rank shit

hex37 committed:
please work memories please

hex37 committed:
bugfixing

also removes diablerie brain transfer

hex37 committed:
Removes Generation code wholesale, fixes obfuscate

hex37 committed:
bugfixing

hex37 committed:
fixing blood bonds and faction saving

hex37 committed:
we didn't need to rename this

hex37 committed:
connection request offer bugfixing

hex37 committed:
fixing it all

hex37 committed:
on my knees begging this works

hex37 committed:
reset character works

hex37 committed:
i reset all savefiles out of spite

hex37 committed:
actually i hate u

hex37 committed:
fix database structure
